src/map/lua/luautils.cpp

```diff
@@ -1933,12 +1933,14 @@ namespace luautils
             return 56;
         }
 
-        CLuaBaseEntity LuaBaseEntity(PTarget);
-        Lunar<CLuaBaseEntity>::push(LuaHandle, &LuaBaseEntity);
+        CLuaBaseEntity LuaBaseEntityTarget(PTarget);
+        CLuaBaseEntity LuaBaseEntityCaster(PCaster);
+        Lunar<CLuaBaseEntity>::push(LuaHandle, &LuaBaseEntityTarget);
+        Lunar<CLuaBaseEntity>::push(LuaHandle, &LuaBaseEntityCaster);
```

Comment:
problem: the lua order used to be (caster, param) but now it's (caster, target, param) - any script that used param will break unless you updated them all (even then, we'd be breaking custom scripts too)

So we should probably just do (caster, param, target) to keep all old scripts working

Reply:
On that note, if you wanted to keep it in the same order as the C++ signature, you could put PCaster after param and have a default value of nullptr

Reply:
Originally it was only taking Target, param. In my head I had added caster as the third option already, so now I am super confused.

Reply:
i mixed up caster and target...

it used to be (target,param) but now it's (target,caster,param).  we can't do that because an existing script will try to use the 2nd argument as "param" but it got moved to the third argument, so they will fail if they try to use it
so we should make it (target,param,caster)

Reply:
Also, since param is an optional field if I leave it where it was and add my new caster to the end, it requires manually calling param or it gets confused and says it tried to convert basentity to integer and freaks out.

Reply:
Can you find a single script that actually uses the param though? I was worried of that too but found none with a git grep and a regular grep against the scripts folder. Like I said, I only found one thing using that params and it was inside a c++.

Reply:
the Lua param order, not the c++ param order